### DISCUSSION TITLE
Selective Style Copying

### DISCUSSION THREAD

**Discussion by Henzxon** [elementor/elementor#21532] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

> I'm always frustrated when i copy a widget style and only want to copy specific styling onto another module. For example, a button style. I want to copy the style into a different module excluding other styles specially transform, rotate, animation delay that are unique most of the time for that element. 
>
>It's destructive, when you copy paste style over a module which overrides all customizations. It's time-consuming to fix something you customized so much.

Imagine that you set an animation delay and custom class for multiple widget so it appears in a wave pattern. With current copy paste, you're animation delay is useless as it will be overriden.

### Describe the solution you'd like

>  It would be nice if I can copy specific widget style settings and apply it to a different module and an option to extent styles to all similar settings (regardless or the widget), elements within the container or page. This ensures that you won't override some widget settings like animation delay and other unique widget customizations for certain widgets.
> 
>  E.g.  Button Padding >> Extent button padding styles
> 
>  Extend Style Options:
> 1. padding style => all buttons  within: parent container or section
> 2. padding style => all buttons on the page
> 3. padding style => throughout the site
> 4. padding style=> page content/body

Other solution would be allowing users to save current module changes as custom style which can be accessed conveniently under a dropdown area within the top of the module.

> **I hope Elementor in the futture supports:**
> - Components and Instances (Figma) implements this really well.
> - Custom style options for modules. (1 button is not applicable on the entire website) and the global feature is less helpful.
> 

### Describe alternatives you've considered

> Apply other style later on but I find it counter-intuitive or use custom class but it deafeats the purpose of the style tab if you cannot use it. I understand that other agencies design using Figma or XD but I design on the fly which others do as well.
> 
> I want to update border radius but the module transform, attributes, custom CSS, gradient overlay and other styling gets copied over as well which is not needed.
> 
>If i set a container animation delay of say, 100,150,200,250 and so on...Once you extend the styles, it will override those unique values. Most of the time we just want to modify certain element styles without affecting unique custom properties like transform, position, attributes and other data.

### Additional context

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/102000469/222751694-e6ffc85a-2541-4df6-8036-0382f994ec51.png">

**Comment by jimmcadory** [comment]
I would love this.  I post a lot of photos and I have to go to each one and set the styling even though it's the same for each.  It's time consuming and could be quickly addressed with this suggestion.

**Comment by jazir555** [comment]
An interesting feature request!

**Comment by ayoubkhan558** [comment]
+1

**Comment by bentographics** [comment]
Exactly what I was looking for as well. I would almost rename it to "Selective Style Paste".

For examply this would be highly useful to apply animations to elements throughout a landing page e.g.
1. For example, I have a scroll animation set up for an image.
2. Now I want to appy that exact animation effect to other widgets e.g. buttons, headlines etc.
3. If I simply use "paste style" from the image as reference element, it'll also apply other settings such as borders, border radius, color etc. 
4. We need a "selective style paste" feature e.g. "Past Animation Effect (only)"
